### ISSUE TITLE
Option to allow (part of) shared data to be always returned

### ISSUE THREAD

**Issue by ajnsn** [inertiajs/inertia-laravel#164]
Sorry for being verbose.

tl;dr: Can we think about an option to always share specified `Inertia::share`-data regardless of the `only` attribute?

If you have a bigger application and want to always respond with some default data, you have to maintain the `only` attribute at lot of places in the frontend. When your shared data changes, you need to follow that up in all those places. 

---

**Example**

As recommend, I am using the `Inertia::share`to share e.g. form errors, notifications/flash messages etc.
For that I have created a dedicated middleware which has the following `handle` method:

```php
...

$sharedData = [
    'auth' => function () use ($request) {
         ...
    ],
    'notification' => function () use ($request) {
        ...
    },
    'errorMessage' => function () use ($request) {
       ...
    },
    'errors' => function () use ($request) {
       ...
    },
];

Inertia::share($sharedData);

return $next($request);
```

When I submit a form e.g. to update an item, my form validation errors did not display.
This is correct as I am using the `only` attribute to only fetch the updated item and the actions:

```javascript
this.$inertia.patch(updateRoute), data, {
    preserveScroll: true,
    only: ['item', 'actions'],
});
```
Instead the `only` attribute needs to look like this:

```javascript
only: ['auth', 'notification', 'errorMessage' , 'errors', 'item', 'actions']
```

My current solution is to modify the `X-Inertia-Partial-Data` header in my middleware to share the data. I simply add all my array keys of my `sharedData` to this header:

```php
...

Inertia::share($sharedData);

if ($request->header('X-Inertia')) {
    $only = array_filter(explode(',', $request->header('X-Inertia-Partial-Data')));
    if (! empty($only)) {
        $request->headers->set('X-Inertia-Partial-Data', implode(',', array_merge(array_keys($sharedData), $only)));
    }
}

return $next($request);
```

**Comment by claudiodekker** [comment]
Related: https://github.com/inertiajs/inertia/issues/205#issuecomment-690047594

**Comment by ajnsn** [comment]
Closing in favor of [#377](https://github.com/inertiajs/inertia/issues/377)
